lib/_tls_wrap.js

```diff
@@ -1085,12 +1085,34 @@ exports.connect = function(/* [port, host], options, cb */) {
     socket.removeListener('end', onHangUp);
   });
 
+  var socketAddress;
+
+  if (options.socket) {
+    options.socket.once('connect', function() {
+      socketAddress = {
+        host: options.socket.remoteAddress,
+        port: options.socket.remotePort
+      };
+    });
+  } else {
+    if (options.path && !options.port) {
+      socketAddress = { path: options.path };
+    } else {
+      socketAddress = {
+        port: options.port,
+        host: options.host,
+        localAddress: options.localAddress
+      };
+    }
+  }
+
```

Comment:
This adds a closure and makes the closure for onHangup() bigger (i.e., more memory, more gc'ing.)  I would like to avoid that.